actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -54,13 +54,21 @@ def callable_cache_key?
       def collection_by_cache_keys
         seed = callable_cache_key? ? @options[:cached] : ->(i) { i }
 
+        key_base = ActiveSupport::Cache::KeyMaster.new(@view.combined_fragment_cache_key(nil)).cache_key
```

Comment:
One thing I don't know here is if the `@view.combined_fragment_cache_key(nil)` can return back any objects that respond to `:cache_version`. If so we need to also output make the keymaster object able to initialize with a default cache key as well. Should be an easy-ish enough.